### PR TITLE
Deduplicate body names using unordered set

### PR DESCRIPTION
**Pull request by rschlaikjer**
Significantly faster for high-body-count scenes - 

Before:
```
2024-08-02 11:39:01,380 openrave [INFO] [jsonreader.cpp:367 JSONReader::ExtractAll] Dedupe time for 7131 bodies: 108'424us
```

After:
```
2024-08-02 11:34:31,265 openrave [INFO] [jsonreader.cpp:367 JSONReader::ExtractAll] Dedupe time for 7131 bodies: 832us
```

### CODE REVIEW

File: src/libopenrave-core/jsonparser/jsonreader.cpp [hunk 1]
```diff
@@ -343,23 +343,20 @@ class JSONReader
...
+                // If this is the first time we've seen this body, add the name to the existing name set and continue
+                if (existingBodyNames.find(bodyName) == existingBodyNames.end()) {
+                    existingBodyNames.emplace(bodyName);
+                    continue;
                 }
-                --iTest;
+
+                // If this body duplicates a body with a higher index, remove it
+                RAVELOG_WARN_FORMAT("env=%d, remove redundant entry %d with body name '%s'", _penv->GetId()%iBody%bodyName);
+                envInfo._vBodyInfos.erase(envInfo._vBodyInfos.begin() + iBody);
```
Comment (rdiankov) [inline src/libopenrave-core/jsonparser/jsonreader.cpp:359]: I wonder if we can speed this up by:
```
std::swap(envInfo._vBodyInfos[iBody], envInfo._vBodyInfos[iValidSize]);
iValidSize--;
```

where iValidSize starts out as `envInfo._vBodyInfos.size()`

Then when the for loop ends, we can resize `envInfo._vBodyInfos` to `iValidSize`

Comment (rschlaikjer) [reply src/libopenrave-core/jsonparser/jsonreader.cpp:359]: Yeah, that should be faster. The duplicated case is I think usually a cold path but doesn't hurt.

**Comment by rdiankov** [comment]
thanks

File: src/libopenrave-core/jsonparser/jsonreader.cpp [hunk 2]
```diff
@@ -344,7 +344,7 @@ class JSONReader
 
         // have to remove any duplicate names, prioritize ones that have higher index
         {
-            std::unordered_set<std::string> existingBodyNames;
+            boost::unordered_set<string_view> existingBodyNames;
```
Comment (cielavenir) [inline src/libopenrave-core/jsonparser/jsonreader.cpp]: @rschlaikjer sorry that I missed this, but why don't we use `std::unordered_set<OpenRAVE::string_view>`?

cc @ntohge

Comment (cielavenir) [reply src/libopenrave-core/jsonparser/jsonreader.cpp]: resolved by adding std::hash, https://github.com/rdiankov/openrave/pull/1412

**Pull request merged**